modules/hivemq-edge-module-plc4x/src/main/java/com/hivemq/edge/adapters/plc4x/impl/AbstractPlc4xAdapter.java

```diff
@@ -135,9 +135,28 @@ public void poll(final @NotNull BatchPollingInput pollingInput, final @NotNull B
     public void start(
             final @NotNull ProtocolAdapterStartInput input, final @NotNull ProtocolAdapterStartOutput output) {
         try {
-            // we do not subscribe anymore as no current adapter type supports it anyway
-            initConnection();
-            output.startedSuccessfully();
+            if (connection == null) {
+                synchronized (lock) {
+                    if (connection == null) {
+                        // we do not subscribe anymore as no current adapter type supports it anyway
+                        if (log.isTraceEnabled()) {
+                            log.trace("Creating new instance of Plc4x connector with {}.", adapterConfig);
+                        }
+                        final Plc4xConnection<T> connection = createConnection();
+                        if(connection.isConnected()) {
+                            protocolAdapterState.setConnectionStatus(CONNECTED);
+                            output.startedSuccessfully();
+                        } else {
+                            protocolAdapterState.setConnectionStatus(ERROR);
+                            output.failStart(new Plc4xException("Unable to connect to device"), "Unable to connect to device");
+                        }
+
+                        this.connection = connection;
```

Comment:
This line is executed even if `output.failStart()` is called. Is this expected behavior?

Reply:
Thanks for catching :)
Fixed.